python/lancedb/table.py

```diff
@@ -392,6 +408,18 @@ def _sanitize_vector_column(data: pa.Table, vector_column_name: str) -> pa.Table
         return data
     if not pa.types.is_list(vec_arr.type):
         raise TypeError(f"Unsupported vector column type: {vec_arr.type}")
+    if len(vec_arr.values) % len(data) != 0:
+        if strict_mode:
+            raise ValueError(
+                f"Vector column {vector_column_name} has variable length vectors"
+            )
+        # not a fixed size list, maybe some NaNs
+        lst_lengths = np.unique([len(lst) for lst in vec_arr])
+        ndims = np.max(lst_lengths)
+        mask = pa.array([len(lst) == ndims for lst in vec_arr])
```

Comment:
@changhiskhan I'll just echo what said about also handling [nan]*len(ndims). Maybe also include a check for any nan.
```
ask = pa.array([len(lst) == ndims and not numpy.isnan(lst).any()for lst in vec_arr])
```

Reply:
I'd rather not introduce numba as dependency. benchmarking this locally, i get ~0.1ms for `np.isnan` on 100,000 element array. So this is unlikely to be a bottleneck